### PR TITLE
Fix license page title color issues

### LINKED ISSUE #108991: License page appBar text not visible
## Steps to Reproduce

1. Execute `flutter run` on the code sample on `web` or `desktop` platform
2. Maximize application window
3. The main license page have proper appBar bg and text color
4. The sub-section for license display have appBar bg and text both in `white` color

**Expected results:** 
Inner sub-section for license display appBar should have same color theme as main page appBar.

**Actual results:** 
Inner sub-section for license display appBar have background fixed to the scaffold color for both `light` and `dark` themes.

<details>
<summary>Code sample</summary>

```dart
// Flutter Packages
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.purple,
        appBarTheme: const AppBarTheme(
          backgroundColor: Colors.deepPurple,
          foregroundColor: Colors.white,
        ),
        useMaterial3: true,
      ),
      home: const MyLicensePage(),
    );
  }
}

class MyLicensePage extends StatelessWidget {
  const MyLicensePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return LicensePage(
      applicationIcon: Container(
        margin: const EdgeInsets.all(10),
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(35),
          boxShadow: [
            BoxShadow(
              color: (Theme.of(context).brightness == Brightness.dark
                      ? Colors.white
                      : Colors.black)
                  .withOpacity(0.69),
              blurRadius: 10,
            ),
          ],
        ),
        width: 150,
        height: 150,
      ),
      applicationName: 'License Page Issue',
      applicationVersion: '-Infinity',
      applicationLegalese: 'Copyright © 2022 OutdatedGuy',
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```bash
https://jpst.it/2VNyh
```

```bash
Analyzing whats_that_anime...                                           
No issues found! (ran in 12.8s)
```

```bash
[√] Flutter (Channel stable, 3.0.5, on Microsoft Windows [Version 10.0.22622.450], locale en-IN)
    • Flutter version 3.0.5 at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (3 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

Checking Android licenses is taking an unexpectedly long time...[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at C:\Users\Dell\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32526.322
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[√] VS Code (version 1.70.0)
    • VS Code at C:\Users\Dell\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.46.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22622.450]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 103.0.5060.134
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 103.0.1264.77

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

<hr />

Before Scrolling the License

image

<hr />

After Scrolling the License

image

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/108991

Complete details in https://github.com/flutter/flutter/issues/108991#issuecomment-1453516439

### Description
1. This fixes the license page title color in the wide layout when `foregroundColor `is provided in the `AppBarTheme`.
2. Fixes an issue with the license page title color in the nested layout for Material 3. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by HansMuller** [comment]
It looks like there is a legitimate presubmit test failure:
```dart
02:36 +4721 ~15: /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter/test/material/about_test.dart: Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.fuchsia)
 ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
 The following TestFailure was thrown running a test:
 Expected: exactly one matching node in the widget tree
   Actual: _TextFinder:<2 widgets with text "AAA" (ignoring offstage widgets): [Text("AAA",
 dependencies: [DefaultSelectionStyle, DefaultTextStyle, MediaQuery]), Text("AAA", debugLabel:
 ((englishLike titleLarge 2014).merge(blackMountainView titleLarge)).copyWith, inherit: false, color:
 Color(0xdd000000), family: Roboto, size: 20.0, weight: 500, baseline: alphabetic, decoration:
 TextDecoration.none, dependencies: [DefaultSelectionStyle, DefaultTextStyle, MediaQuery])]>
    Which: is too many
 When the exception was thrown, this was the stack:
 #4      main.<anonymous closure> (file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter/test/material/about_test.dart:891:5)
 <asynchronous suspension>
 <asynchronous suspension>
 (elided one frame from package:stack_trace)
 This was caught by the test expectation on the following line:
   file:///Volumes/Work/s/w/ir/x/w/flutter/packages/flutter/test/material/about_test.dart line 891
 The test description was:
   Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant:
   TargetPlatform.fuchsia)
 ════════════════════════════════════════════════════════════════════════════════════════════════════
 02:36 +4721 ~15 -1: /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter/test/material/about_test.dart: Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.fuchsia) [E]
   Test failed. See exception logs above.
   The test description was: Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.fuchsia)
   
 To run this test again: /Volumes/Work/s/w/ir/x/w/flutter/bin/cache/dart-sdk/bin/dart test /Volumes/Work/s/w/ir/x/w/flutter/packages/flutter/test/material/about_test.dart -p vm --plain-name 'Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.fuchsia)'
```

File: packages/flutter/lib/src/material/about.dart [hunk 1]
```diff
@@ -1227,16 +1233,18 @@ class _MasterDetailFlowState extends State<_MasterDetailFlow> implements _PageOp
...
-                title: widget.title,
-                automaticallyImplyLeading: widget.automaticallyImplyLeading,
-                masterViewBuilder: widget.masterViewBuilder,
-              ),
-      ),
+      builder: (BuildContext c) {
+        return BlockSemantics(
+          child: _MasterPage(
+            leading: widget.automaticallyImplyLeading && Navigator.of(context).canPop()
+              ? BackButton(onPressed: () => Navigator.of(context).pop())
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/about.dart:1240]: Prefer `=>` for one-liners that return a value. This is just about the same number of characters:
```suggestion
              ? BackButton(onPressed: () { Navigator.of(context).pop(); })
```

File: packages/flutter/lib/src/material/about.dart [hunk 2]
```diff
@@ -935,29 +941,29 @@ class _PackageLicensePageState extends State<_PackageLicensePage> {
...
-    final Color? color = Theme.of(context).appBarTheme.foregroundColor;
     final TextStyle? effectiveTitleTextStyle = titleTextStyle ?? theme.titleLarge;
-
     return Column(
       mainAxisAlignment: MainAxisAlignment.center,
       crossAxisAlignment: CrossAxisAlignment.start,
       children: <Widget>[
-        Text(title, style: effectiveTitleTextStyle?.copyWith(color: color)),
-        Text(subtitle, style: theme.titleSmall?.copyWith(color: color)),
+        Text(title, style: effectiveTitleTextStyle?.copyWith(color: foregroundColor)),
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/about.dart:965]: Glad to see the foreground color make an appearance here :-)

**Review by HansMuller — COMMENTED** [review]
This looks good. Unfortunately there's a presubmit failure to contend with.

**Comment by TahaTesser** [comment]
> It looks like there is a legitimate presubmit test failure:

 This doesn't fail on macOS locally, looks like other platforms show lateral UI when not configuring surface size. 
 I've updated the tests to make sure we get the nested UI regardless of the platform for these nested UI related tests.

**Comment by TahaTesser** [comment]
```
: Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.fuchsia)
 ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
```

This is an odd one.  The Scrollbars test doesn't fail on macOS or Windows for me locally. I will take a look on Monday.


No issue locally, even for ` TargetPlatform.fuchsia)`

> ✓ Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.android)
> ✓ Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.fuchsia)
> ✓ Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.iOS)
> ✓ Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.linux)
> ✓ Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.macOS)
> ✓ Conflicting scrollbars are not applied by ScrollBehavior to _PackageLicensePage (variant: TargetPlatform.windows)

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**